client/rpc-java-client/src/main/java/org/sonarsource/sonarlint/core/rpc/client/SonarLintRpcClientDelegate.java

```diff
@@ -169,6 +169,10 @@ default void didReceiveServerHotspotEvent(DidReceiveServerHotspotEvent params) {
   String matchSonarProjectBranch(String configurationScopeId, String mainBranchName, Set<String> allBranchesNames,
     SonarLintCancelChecker cancelChecker) throws ConfigScopeNotFoundException;
 
+  default boolean matchProjectBranch(String configurationScopeId, String branchNameToMatch, SonarLintCancelChecker cancelChecker) throws ConfigScopeNotFoundException {
```

Comment:
You could also mark this method as Deprecated, this is what will make Java clients aware